### PR TITLE
docs: extend docstrings helicity and kinematics

### PR DESCRIPTION
**Pull request by redeboer**
Added several examples that explain the implementations in the `helicity` and `kinematics` modules.

A preview of the API can be viewed here:
https://ampform--100.org.readthedocs.build/en/100/api/ampform.html

### CODE REVIEW

**Comment by redeboer** [comment]
@spflueger @Leongrim I think there is something off with `formulate_clebsch_gordan_coefficients`, as it doesn't agree with [Kutschke 1996](https://ampform.readthedocs.io/en/0.10.3/references.html#id8). The problem can best be seen in the new docstring [here](https://ampform--100.org.readthedocs.build/en/100/api/ampform.helicity.html#ampform.helicity.formulate_clebsch_gordan_coefficients)

For reference, here is the current implementation:
https://github.com/ComPWA/ampform/blob/b4f6b33be7b7dce740dcf834aa10f2f536f06c1d/src/ampform/helicity/__init__.py#L364-L380

**Comment by spflueger** [comment]
> @spflueger @Leongrim I think there is something off with [`formulate_clebsch_gordan_coefficients`](https://ampform.readthedocs.io/en/0.10.3/api/ampform.helicity.html#ampform.helicity.formulate_clebsch_gordan_coefficients), as it doesn't agree with [Kutschke 1996](https://ampform.readthedocs.io/en/0.10.3/references.html#id8). The problem can best be seen in the new docstring [here](https://ampform--100.org.readthedocs.build/en/100/api/ampform.helicity.html#ampform.helicity.formulate_clebsch_gordan_coefficients).
> 
> For reference, here is the current implementation:
> https://github.com/ComPWA/ampform/blob/b4f6b33be7b7dce740dcf834aa10f2f536f06c1d/src/ampform/helicity/__init__.py#L364-L380

Could you summarize the problem for me. I don't understand whats wrong, so some example showing an input an output and what output you expect

**Comment by redeboer** [comment]
Problem has been (partially) solved: AmpForm does not use the canonical formalism directly, but uses [Eq. (4.32) in Chung 2014](https://suchung.web.cern.ch/spinfm1.pdf#page=18) to compute the canonical basis from the helicity basis (hence `formalism="canonical-helicity"`).

This also explains the problems experienced in https://github.com/ComPWA/qrules/pull/84, namely that nodes on a `StateTransition` produced with canonical(-helicity) formalism do not comply with `spin_conservation` (the spin projections are actually helicities and are not conserved in the transformation to canonical).

I extended the docstring to `formulate_clebsch_gordan_coefficients` with more info a links, hope this reduces any potential future confusion.

**Comment by redeboer** [comment]
@spflueger, there is still one thing in the implementation that bothers me. The example in the [new docstring](https://ampform--100.org.readthedocs.build/en/100/api/ampform.helicity.html#ampform.helicity.formulate_clebsch_gordan_coefficients) follows [Eq. (4.32) in Chung 2014](https://suchung.web.cern.ch/spinfm1.pdf#page=18) and correctly reproduces the output from `formulate_clebsch_gordan_coefficients`:
image

The implementation is different though. Why is angular momentum **projection** used there as well?
https://github.com/ComPWA/ampform/blob/b4f6b33be7b7dce740dcf834aa10f2f536f06c1d/src/ampform/helicity/__init__.py#L364-L380

**Comment by spflueger** [comment]
I think all angular momentum projections that are inserted into that CG function call are just zero. I don't know if it has some benefit writing it in this general way though...

**Comment by redeboer** [comment]
> I think all angular momentum projections that are inserted into that CG function call are just zero. I don't know if it has some benefit writing it in this general way though...

Ok, so I propose setting m1 to 0 then, because we now quote that formula in Chung. Indeed it seems that QRules always sets `l_projection` to 0, but we shouldn't rely on that.

**Comment by redeboer** [comment]
https://github.com/ComPWA/ampform/pull/100#issuecomment-886466757
Actually now I notice there is this (unexplained) exception:
https://github.com/ComPWA/ampform/blob/adad83327211342f63f38c79752c6e687fd413e5/src/ampform/helicity/__init__.py#L394-L395
